### ISSUE TITLE
multiple savefig()  --> Error: alloc: invalid block

### ISSUE THREAD

**Issue by picibucor** [matplotlib/matplotlib#4013]
Hi,

I could not have multiple savefig() 's with different styles.

I wrote a sample code on stackoverflow.

http://stackoverflow.com/questions/28024050/multiple-savefig-s-with-different-styles/28024420?noredirect=1#comment44438695_28024420

I got some help somebody posted his results (as a picture). If I copy-paste his code, and run it I got an "alloc: invalid block"  error. And the following 2 pictures:

![result_dark](https://cloud.githubusercontent.com/assets/1676457/5802533/38311550-9ff6-11e4-87af-e1cc5c09afdc.png)

![result_white](https://cloud.githubusercontent.com/assets/1676457/5802534/3f4baba2-9ff6-11e4-96ce-9a3fe9d79c26.png)

The color of labels and legendtext is still white.. 
Is it possible that it is a bug? 

My system:  Win7 + Python3.4 + MatPlotlib 1.4.2

Thx for helping

**Comment by tacaswell** [comment]
In the future, please move code over to gh from SO, given the level of (some times capricious) moderation over there those links are not stable.  

This is a design feature, not a bug.  Almost all of the values controlled by `rcparams` are set at object creation time, not a draw time, because having what your figure will look like when you render it depend on global state is terrifying.   This also allows you to use context managers for the rcparams.  I suggest using something like:

```python
from matplotlib.style import context

def my_plot_function(ax, data, style):
    # do all of your plotting in here, should be fast as no computation
    pass

with context('dark'):
    fig, ax = plt.subplots(1, 1)
    my_plot_function(ax, data, style)
    fig.savefig(...)

with context('fivethirtyeight'):
    fig, ax = plt.subplots(1, 1)
    my_plot_function(ax, data, style)
    fig.savefig(...)

```


[cross posted to SO as well]

**Comment by tacaswell** [comment]
Can you make a new issue with the alloc issue with a minimal example which reproduces the bug?  It is best to not mix the two aspects of this up.

**Comment by picibucor** [comment]
:+1: Thx, it works like a charm.

I thought these issues were just one.
The alloc issue seems to be editor dependent. It does not appear in IDLE.

**Comment by ZhuFengdaaa** [comment]
My editor is Sublime and I have the same problem, my operating system is Windows, do you have any ideas?

**Comment by picibucor** [comment]
I do not run this code in an editor. I wrote a GUI where clicking on a button shows these diagramms.
I wasn't interested in this problem after the diagrams were generated.

**Comment by jenshnielsen** [comment]
To me this sounds like it is backend specific. I think the apparent editor dependency is just caused by a different backed being selected. Googleing the error message seems to indicate that this is TK related. I would guess something is wrong with your TK build. 

If you don't care about which backend you use and have others installed you can select a different one with. 

```python
matplotlib.use('backend')
```

where backend is one of the valid backends such as 'QT4Agg', 'QT5Agg', 'WXAgg', 'GTKAgg' etc.

**Comment by espdev** [comment]
I had the same crash (`matplotlib alloc: invalid block`) in a similar code:
```python
def save_result_figure(image, curve, fname):
    fig, ax = plt.subplots(1, 1)
    ax.imshow(image, cmap='gray')
    ax.plot(curve.x, curve.y, linewidth=2, color='r')
    fig.savefig(fname)
```
This function was called multiple times from the unit tests.

I have added this line to the end of my function:
```python
plt.close()
```

And this error doesn't occur anymore! :)

**Comment by ganeshkp** [comment]
Thank you..plt.close() works..

**Comment by abubelinha** [comment]
Yes, plt.close() solved my issue (Python 3.4 , Windows)
Thank you so much
@abubelinha
